src/main/scala/viper/gobra/translator/encodings/preds/PredEncoding.scala

```diff
@@ -46,6 +46,7 @@ class PredEncoding extends LeafTypeEncoding {
     * (1) exclusive operations on T, which includes literals and default values
     *
     * [Q{d1, ..., dk}: pred(S)] -> make_S_ID([d1], ..., [dk]) where ID is the ID for the pattern used in Q{d1, ..., dk}
+    * [dflt(Pred(ts)°)] -> default() where default is a 0-ary function in the domain
```

Comment:
I have some troubles imagining where a default value for a `Pred` type shows up. Can you point me at an instance in one of the test cases?
Is it correct that you cannot do much with the default value since e.g. fold/unfold does not work thanks to our restriction that you have to write a pred constructor (instead of just a predicate expression)?

Reply:
I believe the default value is pretty much useless per se but it is required for cases where you create a collection of predicates. For example, `x := [5]pred(){}`. Without this, it would crash, as the array encoding first initializes the array with the default value